core/modules/file/views/views_handler_field_file_icon.inc

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * @file
+ * Definition of views_handler_field_file_icon.
+ */
+
+/**
+ * @ingroup views_field_handlers
+ */
+class views_handler_field_file_icon extends views_handler_field {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function option_definition() {
+    $options = parent::option_definition();
+    $options['icon_size'] = array('default' => 32);
+    $options['display_type'] = array('default' => 'svg');
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function options_form(&$form, &$form_state) {
+    parent::options_form($form, $form_state);
+    $form['icon_size'] = array(
+      '#type' => 'number',
+      '#title' => t('Icon size'),
+      '#min' => 8,
+      '#max' => 300,
+      '#default_value' => $this->options['icon_size'],
+      '#field_suffix' => 'px',
+    );
+    $form['display_type'] = array(
+      '#type' => 'radios',
+      '#title' => t('Display type'),
+      '#options' => array(
+        'svg' => t('Inline SVG markup'),
+        'img' => t('Regular image'),
+      ),
+      '#default_value' => $this->options['display_type'],
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render($values) {
+    $value = $this->get_value($values);
+
+    // If there is an extension.
+    if ($extension = pathinfo($value, PATHINFO_EXTENSION)) {
+      $icon_name = $this->mapExtensionToName($extension);
+      if ($this->options['display_type'] == 'img') {
+        $icon = $this->renderImg($icon_name);
+      }
+      else {
+        $icon = $this->renderSvg($icon_name);
+      }
+      return $icon;
+    }
+  }
+
+  /**
+   * Render the icon as svg tag.
+   */
+  protected function renderSvg($icon_name) {
+    $options = array(
+      'attributes' => array(
+        'width' => $this->options['icon_size'],
+        'height' => $this->options['icon_size'],
+      ),
+    );
+    return icon($icon_name, $options);
+  }
+
+  /**
+   * Render the icon as img tag.
+   */
+  protected function renderImg($icon_name) {
+    $attributes = array(
+      'width' => $this->options['icon_size'],
+      'height' => $this->options['icon_size'],
+      'alt' => '',
```

Comment:
Should we add an `alt` text here for a11y? Something along the lines of:
```suggestion
      'alt' => t('File icon for @extension extension', array('@extension' => ...)),
```

It could be the same `alt` text for the svg render above.

Reply:
I belief, an automatic alt text is worse than no text in this case - for A11Y.
It's pure decoration which doesn't provide extra value IMO. Of course, I could be wrong - I'm no A11Y expert.
But in the documentation I read, icons are usually described as "decorative without informational value".

Reply:
@laryn suggested to add a checkbox to toggle alt text on/off. In that case, the admin can decide, whether it's better to add it or not (based on context).

@klonos I used your text suggestion.